File: frontend/src/lib/types/topic.ts
export type TopicStatus = 'active' | 'closed' | 'archived';

export interface Topic {
	id: string;
	default_title: string;
	status: TopicStatus;
	created_at: string;
	closed_at: string | null;
	scoped_title: string | null;
}

export interface TopicCreateResponse {
	topic: Topic;
	token: string;
	magic_link: string;
}

File: backend/app/schemas/topic.py
import uuid
from datetime import datetime

from pydantic import BaseModel, EmailStr, Field

from app.models.enums import TopicStatus


class TopicCreate(BaseModel):
    default_title: str = Field(min_length=1, max_length=200)
    creator_email: EmailStr | None = None


class TopicResponse(BaseModel):
    id: uuid.UUID
    default_title: str
    status: TopicStatus
    created_at: datetime
    closed_at: datetime | None = None
    scoped_title: str | None = None


class TopicCreateResponse(BaseModel):
    topic: TopicResponse
    magic_link: str
